lib/pluginmanager/install.rb

```diff
@@ -43,6 +43,7 @@ def validate_cli_options!
   # Check if the specified gems contains
   # the logstash `metadata`
   def verify_remote!(gems)
+    Gem.sources = Gem::SourceList.from(gemfile.gemset.sources)
     gems.each do |plugin, version|
       puts("Validating #{[plugin, version].compact.join("-")}")
       signal_error("Installation aborted, verification failed for #{plugin} #{version}") unless LogStash::PluginManager.logstash_plugin?(plugin, version)
```

Comment:
Yes, it terminates. Why not using another PR, so make this cleaner? I don't like the idea of doing lot of refactoring, changing the scope of the PR. What do you think?